catalog/dags/data_refresh/remap_table_indices.py

```diff
@@ -0,0 +1,137 @@
+"""
+# Remap Table Indices
+
+This file contains TaskGroups related to rmampping the table indices from the live media table onto the temp table to prepare it for promotion.
+"""
+
+import logging
+from typing import NamedTuple
+
+from airflow.decorators import task, task_group
+from airflow.utils.trigger_rule import TriggerRule
+
+from common.sql import fetch_all, run_sql
+from data_refresh import queries
+
+
+logger = logging.getLogger(__name__)
+
+
+class TableIndex(NamedTuple):
```

Comment:
Question: is the usage of `NamedTuple` instead of dataclasses due to a limitation in Airflow? I seem to recall limitations in task return types. Otherwise, do you have a preference for NamedTuples over dataclasses? And if so, I'd be curious to know the reason :slightly_smiling_face:

Reply:
I believe anything json serializable can be passed through XCOMs, if I'm not mistaken a dataclass would work. No strong preference beyond following convention in the project, and given that we don't require anything more complex. I don't think the pros/cons of either option are particularly significant in this use case.

Reply:
+1 to what Staci said, according to [the documentation](https://airflow.apache.org/docs/apache-airflow/stable/core-concepts/xcoms.html):

> They can have any (serializable) value

I'm assuming "serializable" in this case means JSON serialization. I was able to [return a `NamedTuple` in the `add_rekognition_labels` DAG](https://github.com/WordPress/openverse/blob/41479444c02b394f991d566a6c0e819b706447be/catalog/dags/data_augmentation/rekognition/add_rekognition_labels.py#L133-L139) and Airflow [hydrated it back into the `NamedTuple` just fine](https://github.com/WordPress/openverse/blob/41479444c02b394f991d566a6c0e819b706447be/catalog/dags/data_augmentation/rekognition/add_rekognition_labels.py#L143). In practice (via the UI), it looks like it serializes it to a list: 

![image](https://github.com/user-attachments/assets/04715bb2-7d9f-4499-8dad-e14f956b1976)

![image](https://github.com/user-attachments/assets/8ca30d2a-3e0f-4894-96ad-2b5cda2fbe00)

Dataclasses, although they have more functionality, are a little heavier-weight. If we essentially just need an object to attach a bunch of data to and reference it more easily/programmatically/in a typed way than a dictionary, `NamedTuple` is preferred. That said, I _did_ just find [this documentation on dataclasses!!](https://airflow.apache.org/docs/apache-airflow/stable/core-concepts/taskflow.html#passing-arbitrary-objects-as-arguments) 🤩:

> As mentioned TaskFlow uses XCom to pass variables to each task. This requires that variables that are used as arguments need to be able to be serialized. Airflow out of the box supports all built-in types (like int or str) and it supports objects that are decorated with `@dataclass` or `@attr.define`.

Reply:
Update: I made an upstream PR to make that clearer here! 
https://github.com/apache/airflow/pull/42045